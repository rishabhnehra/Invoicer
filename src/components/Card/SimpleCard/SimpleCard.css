.SimpleCard {
    border-radius: 10px;
    padding: 16px 24px;
    color: #FFF;
    width: 320px;
    background-color: #9E579D;
    box-shadow: 0px 0px 6px 3px rgba(0,0,0,0.1);
}

.SimpleCard h1 {
    margin: 0;
    font-size: 20px;
    font-weight: 600;
}

.SimpleCard p {
    font-size: 11px;
    font-weight: 400;
}

.SimpleCard .actions {
    display: flex;
    justify-content: flex-end;
}

.SimpleCard a {
    color: #FFF;
    font-size: 11px;
    font-weight: 400;
}