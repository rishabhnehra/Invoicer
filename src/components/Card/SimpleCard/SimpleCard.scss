.SimpleCard {
    border-radius: 10px;
    padding: 16px 24px;
    color: #FFF;
    width: 320px;
    background-color: #9E579D;
    box-shadow: 0px 0px 6px 3px rgba(0,0,0,0.1);


    h1 {
        font-size: 20px;
        font-weight: 700;
        margin-bottom: 12px;
    }

    p {
        font-size: 11px;
        font-weight: 500;
        margin-bottom: 12px;
    }

    .actions {
        display: flex;
        justify-content: flex-end;
    }

    a {
        color: #FFF;
        font-size: 11px;
        font-weight: 400;
        text-decoration: none;
    }
}