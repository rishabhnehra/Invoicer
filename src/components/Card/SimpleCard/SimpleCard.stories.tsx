import {Story, Meta} from '@storybook/react/types-6-0';

import {SimpleCard, SimpleCardProps} from './SimpleCard';

export default {
    title: 'Components/Card',
    component: SimpleCard
} as Meta;

const Template: Story<SimpleCardProps> = (args) => <SimpleCard {...args} />

export const Default = Template.bind({});
Default.args = {
    header: 'Create Bill',
    description: `This will initiate the flow of bill creation where
    you\'ll enter the required information and share
    it with your clients either by mail, WhatsApp or
    simple PDF`,
    actionLink: '#',
    actionLabel: 'Click to get started'
}