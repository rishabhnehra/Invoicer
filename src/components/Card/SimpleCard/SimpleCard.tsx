import './SimpleCard.css'

export type SimpleCardProps = {
    header: string,
    description: string,
    actionLabel: string,
    actionLink: string,
}

export const SimpleCard: React.FC<SimpleCardProps> = ({
    header,
    description,
    actionLabel,
    actionLink
}) => {
    return <div className="SimpleCard">
        <h1>{header}</h1>
        <p>{description}</p>
        <div className="actions">
            <a href={actionLink}>{actionLabel}</a>
        </div>
    </div>
}