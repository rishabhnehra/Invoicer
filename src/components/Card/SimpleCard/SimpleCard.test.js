import renderer from 'react-test-renderer';
import { SimpleCard } from './SimpleCard';

describe("<SimpleCard />", () => {
    it("snapshot check", () => {
        const tree = renderer.create(<SimpleCard 
            header='Create Bill'  
            description={`This will initiate the flow of bill creation where
            you\'ll enter the required information and share
            it with your clients either by mail, WhatsApp or
            simple PDF`}
            actionLink='#'
            actionLabel='Click to get started'
        />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});