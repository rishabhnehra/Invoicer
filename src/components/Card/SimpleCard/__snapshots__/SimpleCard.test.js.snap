// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleCard /> snapshot check 1`] = `
<div
  className="SimpleCard"
>
  <h1>
    Create Bill
  </h1>
  <p>
    This will initiate the flow of bill creation where
            you'll enter the required information and share
            it with your clients either by mail, WhatsApp or
            simple PDF
  </p>
  <div
    className="actions"
  >
    <a
      href="#"
    >
      Click to get started
    </a>
  </div>
</div>
`;
